### PR TITLE
changefeedccl: Do not require rangefeed when running initial scan only.

### LINKED ISSUE #99470: changefeedccl: Do not require `kv.rangefeed.enabled` for `initial_scan=only`
We should not require `kv.rangefeed.enabled` setting to be turned on
when running with `initial_scan=only`.  Doing so makes exports needlessly more expensive.

Jira issue: CRDB-25892

Epic CRDB-19123

### PR DESCRIPTION
**Pull request by miretskiy**
Fixes #99470

Release note: None

### CODE REVIEW

**Review by jayshrivastava — APPROVED** [review]

File: pkg/BUILD.bazel [hunk 1]
```diff
@@ -3383,8 +3383,8 @@ test_suite(
     name = "ccl_tests",
     tags = [
         "-broken_in_bazel",
-        "ccl_test",
         "-integration",
+        "ccl_test",
```
Comment (jayshrivastava) [inline pkg/BUILD.bazel:3387]: Interesting...

Comment (miretskiy) [reply pkg/BUILD.bazel:3387]: I don't understand this... just keeps insisting on rearranging it....

Comment (miretskiy) [reply pkg/BUILD.bazel:3387]: Just running ./dev/lint

**Pull request merged**